packages/component-test-utils-react/src/dispatcher/index.js

```diff
@@ -35,18 +34,35 @@ class Dispatcher {
     const hookIndex = this._getHookIndex();
 
     if (this._firstCall) {
-      this.states.push(initialState);
+      this._hookMemo.push(initialState);
```

Comment:
Are you sure of this? I mean, you're pushing this initialstate, not a hook of memo?

Reply:
Not sure at all.

It's all about the hook storage